### PR TITLE
Distinguish "Creator" badge on comments

### LINKED ISSUE #1615: An OP indicator badge in comment
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

it would be nice to have some kind of indicator that shows when the OP responds to a comment on its own post to differentiate it from other commenters.

like this...

Screenshot_20230626-170403

### PR DESCRIPTION
**Pull request by alectrocute**
Hi Lemdevs!

In this PR:

- Make "Post Creator" badge stand out more by:
        - Use "OP" shorthand label by default
        - Add tooltip that says "Creator" upon hover
        - Use `.text-info` color

Screenshot 2023-06-26 at 4 45 50 PM

This is a really important distinguishing badge on comments. I personally dislike the term "Original Poster", acronym "OP", but it has two advantages here: is shorter, takes up less room on row and is very familiar. So with that said, I agree with #1615. Debate on.

Resolves #1615.

Thanks!

### CODE REVIEW

**Review by SleeplessOne1917 — APPROVED** [review]
LGTM, will wait for other feedback before merging. Is the translation for OP merged already?

**Comment by alectrocute** [comment]
> LGTM, will wait for other feedback before merging. Is the translation for OP merged already?

Nope, but I'll open a PR!

File: src/shared/components/comment/comment-node.tsx
Comment (jsit) [inline src/shared/components/comment/comment-node.tsx:318]: This is probably more like a `<span>`. Bootstrap says `<span>` is ok: https://getbootstrap.com/docs/5.3/components/badge/#examples

**Review by jsit — CHANGES_REQUESTED** [review]

**Comment by alectrocute** [comment]
We’ll also need to make sure the OP/creator badge appears on mobile viewport widths, too.

File: src/shared/components/comment/comment-node.tsx
Comment (alectrocute) [reply src/shared/components/comment/comment-node.tsx:318]: Good call.

**Comment by SleeplessOne1917** [comment]
> We’ll also need to make sure the OP/creator badge appears on mobile viewport widths, too.

There are settings in both firefox and chromium devtools that allow you to mimic the screen size of a mobile device. For example, on firefox:
image

**Comment by alectrocute** [comment]
> > We’ll also need to make sure the OP/creator badge appears on mobile viewport widths, too.
> 
> There are settings in both firefox and chromium devtools that allow you to mimic the screen size of a mobile device. For example, on firefox: image

Yep, fully intend on testing it this way. Just leaving a note for myself when I return to this PR later today or tomorrow. 😅

**Comment by dessalines** [comment]
One thing I do on jerboa (that I nabbed from boost for reddit), is to use a badge on the username itself for post creators:

Screenshot_2023-06-27-07-50-30-008_com jerboa

Up to you, but that's also an option.

**Comment by dessalines** [comment]
I merged the translation also, because whatever solution we go with, it could work with tippy.

**Review by dessalines — APPROVED** [review]

**Comment by alectrocute** [comment]
@jsit Made your suggested changes!

@dessalines Fixed the issue with it not appearing on mobile.

Screenshot 2023-06-27 at 11 42 47 AM

Screenshot 2023-06-27 at 11 42 44 AM

It's lowercase in my screenshots because I forgot to pull the submodules.

We should be good to go!

**Review by SleeplessOne1917 — CHANGES_REQUESTED** [review]
I think the OP badge should be uppercase. It looks like a word and not an acronym when written lowercase.

**Comment by alectrocute** [comment]
> I think the OP badge should be uppercase. It looks like a word and not an acronym when written lowercase.

See my latest comment:

> It's lowercase in my screenshots because I forgot to pull the submodules.

**Review by SleeplessOne1917 — APPROVED** [review]

File: src/shared/components/comment/comment-node.tsx
Comment (jsit) [inline src/shared/components/comment/comment-node.tsx:319]: I don't think we need `d-sm-inline` here anymore

**Review by jsit — CHANGES_REQUESTED** [review]

**Comment by jsit** [comment]
I know this isn't the fault of this PR, but it seems weird that we're hiding the other badges on mobile? That seems like really useful/important info.

**Comment by jsit** [comment]
> > I think the OP badge should be uppercase. It looks like a word and not an acronym when written lowercase.
> 
> See my latest comment:
> 
> > It's lowercase in my screenshots because I forgot to pull the submodules.

I did `git submodule update` and am still seeing lowercase.

**Comment by alectrocute** [comment]
> I know this isn't the fault of this PR, but it seems weird that we're hiding the other badges on mobile? That seems like really useful/important info.

Yeah, now that I think about it – These badges are really critical. We should really address this immediately by removing `d-none d-sm-inline` from all these role badges.

> I did git submodule update and am still seeing lowercase.

Weird. I'll just add a `.toUpperCase()` proto, because what the heck. It's not a big deal I suppose.

**Comment by SleeplessOne1917** [comment]
Maybe we could save some space by doing something like  what Hexbear does and using letter badges.

Admin:
Screenshot_2023-06-27-13-55-45-60_50ecd26a21c2c56ce608de5e94733463

Mod:
Screenshot_2023-06-27-13-57-51-48_50ecd26a21c2c56ce608de5e94733463

**Comment by alectrocute** [comment]
@SleeplessOne1917 @jsit Please see my latest changes to this branch. I refactored the whole thing. 😅

I like the idea of using letters to save space. I say let's do this! @jsit thoughts?

**Comment by alectrocute** [comment]
Pushed yet another refactor. Let me know what you all think!

**Comment by alectrocute** [comment]
Walkthrough video of most recent changes:

[Untitled.webm](https://github.com/LemmyNet/lemmy-ui/assets/35377827/e9a7a588-b262-430a-a3e6-ffa2719f0e7e)

**Comment by alectrocute** [comment]
Screenshot 2023-06-27 at 3 02 42 PM

Most recent changes, as per chat on matrix.

**Review by SleeplessOne1917 — APPROVED** [review]

**Review by jsit — APPROVED** [review]

**Pull request merged**
